pump/storage/sorter.go

```diff
@@ -124,6 +124,16 @@ func (s *sorter) setResolver(resolver func(startTS int64) bool) {
 	s.resolver = resolver
 }
 
+func (s *sorter) allMatched() bool {
+	var match bool
+
+	s.lock.Lock()
+	match = len(s.waitStartTS) == 0
+	s.lock.Unlock()
+
+	return match
```

Comment:
How about using `defer` and return the bool expression directly?

Reply:
https://github.com/pingcap/tidb-binlog/pull/701/commits/9c56122d3a6ad56ab107022265301ac87ea28b73